### DISCUSSION TITLE
Is match decryption supposed to be this slow?

### DISCUSSION THREAD

**Discussion by ryechus** [fastlane/fastlane#17906] [Q&A]
Hi,

I am noticing that match is taking 10 minutes to decrypt the store. Is this normal? I am using S3 for storage. I have over 1000 certificates and provisioning profiles.

**Comment by rogerluan** [comment]
Hey @ryechus 👋 

Not sure, but 600ms per item doesn't sound that much, given that we encrypt/decrypt every single item.

Maybe this could be improved by encrypting/decrypting the entire directory, but that'd mean changing how the items are stored (we'd have to store e.g. 1 binary with many of profiles/certificates inside of it, instead of storing many individual items 🤔 ). Not sure about the implications of that, but at least for s3 and google cloud that doesn't sound too much for a problem.
